### PR TITLE
Persist Roles When someone leaves and rejoins quickly

### LINKED ISSUE #493: Persist a user's roles when they leave and join back quickly

### PR DESCRIPTION
**Pull request by tanndlin**
fixes #493

### CODE REVIEW

**Comment by tanndlin** [comment]
This isn't persisted in the Database, the main goal of this is to give the roles back when someone leaves and joins back relatively quickly. Mainly to stop role-avoiding

File: src/app/handlers/new_member_role.handler.ts [hunk 1]
```diff
@@ -7,6 +7,12 @@ export class NewMemberRoleHandler implements IHandler {
   constructor(public container: IContainer) {}
 
   public async execute(member: GuildMember): Promise<void> {
+    const persistedRoles = this.container.userService.getPersistedRoles(member.id);
+    if (persistedRoles) {
+      await member.roles.add(persistedRoles);
+      return; // No need to check the rest if they were already in the server
+    }
```
Comment (c650) [inline src/app/handlers/new_member_role.handler.ts:14]: sketchy to have this not be in its own handler

**Review by c650 — APPROVED** [review]
lgtm!

File: src/app/handlers/new_member_role.handler.ts (hunk 1 above)
Comment (tanndlin) [reply src/app/handlers/new_member_role.handler.ts:14]: ok, will do

**Pull request merged**